docs/models/selector.md

```diff
@@ -0,0 +1,26 @@
+<!-- 
+    This document has been automatically generated.
+    It should NOT BE EDITED.
+    To update this part of the documentation,
+    please type the following from the repository root:
+    $ make docs-build
+-->
+# Model Selector
+
+Selects the matching model for a given event.
+
+    Attributes:
+        model_rules (dict): Stores the list of rules for each model.
+        decision_tree (dict): Stores the rule checking order for model selection.
+    
+
+## Decision Tree
+
+    If event.event_source == browser:
+        If event.event_type == page_close:
+            PageCloseBrowserEventModel
+    If event.event_source == server:
+        If event.event_type == event.context.path:
+            ServerEventModel
+
+
```

Comment:
Good Point) Thanks) Yes indeed, the documentation for `ModelSelector` is much better documented using `mkdocstring`)

Updated the PR to use `mkdocstring` for the `ModelSelector` documentation.

The only part that is missing is the decision tree visualization which is made with the `generate_docs` script.
I think it might be useful to be able to see what happens with the decision tree when we add a new model ? WDYT ?